libs/fn/src/lib/button/button.component.ts

```diff
@@ -28,70 +29,116 @@ export type ButtonType = '' | 'secondary' | 'flat' | 'link' | 'outline' | 'naked
-    private _subscriptions = new Subscription();
+    /**
+     * Set emphasized state of the button
+     */
+    @Input()
+    @coerceBoolean
+    emphasized: boolean;
 
-    /** @hidden */
-    constructor(private _elementRef: ElementRef, private _changeDetectorRef: ChangeDetectorRef) {
-        super();
+    /**
+     * Set selected state of the button
+     */
+    @Input()
+    set selected(value: BooleanInput) {
+        this.setSelected(coerceBooleanProperty(value));
     }
 
-    /** Function runs when component is initialized
-     * function should build component css class
-     * function should build css style
-     */
-    public ngOnChanges(): void {
-        this.buildComponentCssClass();
+    get selected(): boolean {
+        return this._selected;
     }
 
-    public ngOnInit(): void {
-        this.buildComponentCssClass();
+    /**
+     * Native disabled attribute of button element
+     */
+    @Input()
+    get disabled(): boolean {
+        return this._disabled || (this.selectComponent !== null && this.selectComponent.disabled);
     }
 
-    /** @hidden */
-    ngOnDestroy(): void {
-        this._subscriptions.unsubscribe();
+    set disabled(value: BooleanInput) {
+        const newDisabledState = coerceBooleanProperty(value);
+        if (this._disabled !== newDisabledState) {
+            this._disabled = newDisabledState;
+            this._changeDetectorRef.markForCheck();
+        }
     }
 
-    @applyCssClass
-    /** CssClassBuilder interface implementation
-     * function must return single string
-     * function is responsible for order which css classes are applied
+    /**
+     * Value of the button
+     */
+    @Input()
+    value: string;
+
+    /**
+     * Additional HTML classes
      */
-    buildComponentCssClass(): string[] {
-        return [
-            'fn-button',
-            this.fnType ? `fn-button--${this.fnType}` : '',
-            this._disabled || this._ariaDisabled ? 'is-disabled' : '',
-            this.glyph && !this.label ? 'fn-button--icon-only' : '',
-            this.class
-        ];
+    @Input()
+    class: string;
+
+    /**
+     * Fiori Next button type class getter
+     */
+    @HostBinding('attr.class')
+    get fnTypeClass(): string {
+        return this.fnType ? `fn-button--${this.fnType}` : '';
+    }
+
+    /** @hidden */
+    _selected: boolean;
+
+    /** @hidden */
+    constructor(
+        @Optional() @Inject(SelectComponentRootToken) private selectComponent: SelectComponentRootToken,
+        private _elementRef: ElementRef,
+        private _changeDetectorRef: ChangeDetectorRef
+    ) {
+        super();
     }
 
     /** HasElementRef interface implementation
      * function used by applyCssClass and applyCssStyle decorators
      */
-    public elementRef(): ElementRef<any> {
+    elementRef(): ElementRef<HTMLButtonElement | HTMLAnchorElement> {
         return this._elementRef;
     }
 
+    /** @hidden */
+    setSelected(selected: boolean): void {
```

Comment:
for every @hidden property/method we use `_` prefix

Reply:
In this case setSelected is part of the abstract class, which can be used across many components. marking all of them with underscore prefix would make api a bit ugly. What do you think?